### ISSUE TITLE
Path scrubbing not working for python 3.8 package installed with homebrew

### ISSUE THREAD

**Issue by R1kk3r** [pantsbuild/pex#1025]
Hi there, I got an issue with a pex file embedding a dependency (protobuf) under OSX. I guess it could be related to the protobuf package and how the package is installed with brew but maybe also in the way the scrubbing is done by PEX.

If I install `protobuf` through `pip3.8 install protobuf` the scrubbing works fine because `protobuf` is inside the path `/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages//google/protobuf/`.

But if you install the `protobuf` brew package through `brew install protobuf`, it is installed under `/usr/local/Cellar/protobuf/` and the python package is registered inside `/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages//homebrew-protobuf.pth` with the following line:
```
import site; site.addsitedir('/usr/local/Cellar/protobuf/3.12.4/libexec/lib/python3.8/site-packages')
```

I put some logs to print `module.__path__` and `site_libs` before `cls._tainted_path(...)` is [called](https://github.com/pantsbuild/pex/blob/master/pex/pex.py#L209). Here is the difference between python3.7 and 3.8.

Python 3.7:
```
Module path: ['/Users/XXX/repack.pex/google', '/usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/google']
Site_libs : {'/usr/local/lib/python3.7/site-packages', '/usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages'}
pex: Scrubbing google.__path__: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/google
```

Python 3.8:
```
Module path: ['/Users/XXX/repack.pex/google', '/usr/local/Cellar/protobuf/3.12.4/libexec/lib/python3.8/site-packages/google']
Site_libs : {'/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages', '/usr/local/lib/python3.8/site-packages'}

```

Notice there is no "Scrubbing" for python 3.8.  It is due to the change in where python package is located. Instead of being inside the python 3.8 directory, it is put as a standard `brew` package. The import is indirectly managed inside the `homebrew-protofbu.pth` file (see above for the path and content)

The `cls._tainted_path(...)` is then not able to to identify that the path matches in python 3.8 since they are completely different.

python 3.7 match
Module path : **/usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages**/google
Site_libs : **/usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages**

python 3.8 no match
Module path : /usr/local/Cellar/protobuf/3.12.4/libexec/lib/python3.8/site-packages/google
Site_libs : /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages

Any solution ?

**Comment by R1kk3r** [comment]
```
python3.8 -m site
sys.path = [
    '/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python38.zip',
    '/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8',
    '/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/lib-dynload',
    '/usr/local/lib/python3.8/site-packages',
    '/usr/local/Cellar/protobuf/3.12.4/libexec/lib/python3.8/site-packages',
    '/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages',
]
```

It seems the issue is maybe how the site-package are retrieved here https://github.com/pantsbuild/pex/blob/master/pex/pex.py#L146 ? 

I don't understand all the magic under the hood, let me know if you have a solution to my problem.

**Comment by jsirois** [comment]
Just responding to your second comment: https://github.com/pantsbuild/pex/issues/1025#issuecomment-683922218
python3.8 -m site prints out the sys.path, which includes `/usr/local/Cellar/protobuf/3.12.4/libexec/lib/python3.8/site-packages` but not because that entry is a "site package". IOW, you should find a similar difference using the following:
```
$ python3.8 -m site
sys.path = [
    '/home/jsirois',
    '/usr/lib/python38.zip',
    '/usr/lib/python3.8',
    '/usr/lib/python3.8/lib-dynload',
    '/home/jsirois/.local/lib/python3.8/site-packages',
    '/usr/lib/python3.8/site-packages',
]
USER_BASE: '/home/jsirois/.local' (exists)
USER_SITE: '/home/jsirois/.local/lib/python3.8/site-packages' (exists)
ENABLE_USER_SITE: True
```

Vs:
```
$ python3.8 -c 'import site; print(site.getsitepackages())'
['/usr/lib/python3.8/site-packages']
```

**Comment by jsirois** [comment]
@R1kk3r can you run your PEX file like so?: `PEX_VERBOSE=9 /my/pex/file ...`

I want to see if this log line triggers for `/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homebrew-protobuf.pth`:
https://github.com/pantsbuild/pex/blob/b51b340bea8a585a246997071a815d008986d6e3/pex/pex.py#L130

**Comment by jsirois** [comment]
Note to self, the relevant docs: https://docs.python.org/3/library/site.html

**Comment by jsirois** [comment]
Actually, it looks like `site.addsitedir` just adds the directory to `sys.path`, It does not lter what `site.getsitepackages()` returns. I think then the solution is to snapshot sys.path prior to executing `.pth` files and then consider any new paths added to `sys.path` post-execution of all such files as taineted.

**Comment by R1kk3r** [comment]
PEX_VERBOSE=9 python 3.7
```
pex: Found site-library: /usr/local/lib/python3.7/site-packages
pex: Found site-library: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages
pex: Not a tainted path element: /Users/XXX/myproject.pex/.bootstrap
pex: Not a tainted path element: /Users/XXX/myproject.pex
pex: Not a tainted path element: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python37.zip
pex: Not a tainted path element: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7
pex: Not a tainted path element: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload
pex: Tainted path element: /usr/local/lib/python3.7/site-packages
pex: Scrubbing from site-packages: /usr/local/lib/python3.7/site-packages
pex: Scrubbing from user site: /Users/XXX/Library/Python/3.7/lib/python/site-packages
pex: Scrubbing google.__path__: /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/google
pex: Activating PEX virtual environment from /Users/XXX/myproject.pex :: Searching depex: Activating PEX virtual environment from /Users/XXX/myproject.pex: 3.3ms         
pex:   Searching dependency cache: /Users/XXX/myproject.pex/.deps: 3.2ms
pex: PYTHONPATH contains:
pex:   * /Users/XXX/myproject.pex/.bootstrap
pex:     /Users/XXX/myproject.pex
pex:   * /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python37.zip
pex:     /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7
pex:     /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload
pex:   * - paths that do not exist or will be imported via zipimport
```

PEX_VERBOSE=9 python 3.8
```
pex: Found site-library: /usr/local/lib/python3.8/site-packages
pex: Found site-library: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages
pex: Not a tainted path element: /Users/XXX/myproject.pex/.bootstrap
pex: Not a tainted path element: /Users/XXX/myproject.pex
pex: Not a tainted path element: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python38.zip
pex: Not a tainted path element: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8
pex: Not a tainted path element: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/lib-dynload
pex: Tainted path element: /usr/local/lib/python3.8/site-packages
pex: Scrubbing from site-packages: /usr/local/lib/python3.8/site-packages
pex: Scrubbing from user site: /Users/XXX/Library/Python/3.8/lib/python/site-packages
pex: Scrubbing google.__path__: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/google
pex: Activating PEX virtual environment from /Users/XXX/myproject.pex :: Searching depex: Activating PEX virtual environment from /Users/XXX/myproject.pex: 3.3ms         
pex:   Searching dependency cache: /Users/XXX/myproject.pex/.deps: 3.3ms
pex: PYTHONPATH contains:
pex:   * /Users/XXX/myproject.pex/.bootstrap
pex:     /Users/XXX/myproject.pex
pex:   * /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python38.zip
pex:     /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8
pex:     /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/lib-dynload
pex:   * - paths that do not exist or will be imported via zipimport
```

**Comment by jsirois** [comment]
@R1kk3r thanks for your response, but that's troubling. There is no `Found .pth file: ...` in either of your outputs and that's unexpected. I'd expect to see `Found .pth file: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/homebrew-protobuf.pth`. 

Also troubling about your output is this line for python 3.8:
```
Scrubbing google.__path__: /usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/google
```

The crux of your original bug post is that this style line was present for python 3.7 but it was not present for 3.8. Can you ensure your environment is set up as it was when you reported the bug and then re-check the python 3.7 and 3.8 cases for the `Found .pth file: ...` line?

**Comment by jsirois** [comment]
@R1kk3r for sanity sake, what version of Pex are you using?

**Comment by jsirois** [comment]
OK @R1kk3r I'm going to assume your original bug post is finding the .pth file since that case is reproduced and fixed in #1026.
If you have a chance, you could test this fixes your problem by cloning this repo / using that PR via:
```
$ python -mpex [rest of pex args you would normally use to build a pex file]
$ PEX_VERBOSE=9 [the runtime pex command line you would normally use here]
```

**Comment by R1kk3r** [comment]
@jsirois My bad, the python 3.8 verbose logs were wrong since I forgot to uninstall `protobuf` in pip3.8. I updated the logs above with the problematic config (`protobuf`) installed with homebrew).
